projects/vdk-core/src/vdk/internal/builtin_plugins/job_properties/datajobs_service_properties.py

```diff
@@ -63,8 +69,29 @@ def get_all_properties(self) -> Dict[str, PropertyValue]:
 
     def set_all_properties(self, properties: Dict[str, PropertyValue]) -> None:
         """
-        Persists the passed properties overwriting all previous properties
+        Invokes the write pre-processors if any such configured.
+        Persists the passed properties overwriting all previous properties.
         """
+        if self._write_preprocessors:
+            properties = deepcopy(
```

Comment:
Why not name this variable properties_copy ? Having 2 variables with the same name but different scope is confusing .

Reply:
Because in case of potential preprocessing modification, we don't want to change user's outer scope `set_all_properties()` parameter, the value modified is expected upon return.
I've added a unit test to cover/visualize this scenario

Reply:
In details - not introducing a new variable, because I want to dereference the param with a deep copy if we're about to modify it, then proceed further with handling;

Reply:
Maybe right more detailed comment "keeps the outer scope originally-referenced dictionary intact" was not enough for me to get it.